indexer/lib/fwss-handlers.js

```diff
@@ -64,18 +65,40 @@ export async function handleFWSSDataSetCreated(
 /**
  * Handle Filecoin Warm Storage Service service termination
  *
- * @param {Env} env
- * @param {any} payload
+ * @param {{
+ *   DEFAULT_LOCKUP_PERIOD_DAYS?: number
+ *   GENESIS_BLOCK_TIMESTAMP?: number
+ *   DB: D1Database
+ * }} env
+ * @param {object} payload
+ * @param {string} payload.data_set_id
+ * @param {string} payload.block_number
  * @throws {Error}
  */
 export async function handleFWSSServiceTerminated(env, payload) {
+  const DEFAULT_LOCKUP_PERIOD_DAYS = env.DEFAULT_LOCKUP_PERIOD_DAYS || 30 // 30 days default
+  const GENESIS_BLOCK_TIMESTAMP = env.GENESIS_BLOCK_TIMESTAMP || 1667326380 // Calibration genesis as fallback
+
+  // Convert block_number (epoch) to Unix timestamp (in seconds)
+  const epochTimestamp = epochToTimestamp(
+    payload.block_number,
+    GENESIS_BLOCK_TIMESTAMP,
+  )
```

Comment:
Why are we needing timestamps here? Couldn't we only use black numbers everywhere exclusively, and not even care about how long one block is (or should be)?